clippy_lints/src/derivable_impls.rs

```diff
@@ -83,6 +83,7 @@ impl<'tcx> LateLintPass<'tcx> for DerivableImpls {
             if !attrs.iter().any(|attr| attr.doc_str().is_some());
             if let child_attrs = cx.tcx.hir().attrs(impl_item_hir);
             if !child_attrs.iter().any(|attr| attr.doc_str().is_some());
+            if !adt_def.is_enum();
```

Comment:
I think this should check for `if adt_def.is_struct();` instead, to make sure that we don't get any FPs on unions, as they can also not derive `Default`. Otherwise, this looks good to me :upside_down_face: